account-kit/react/src/components/auth/sections/EmailAuth.tsx

```diff
@@ -28,8 +28,15 @@ export const EmailAuth = memo(
     const signer = useSigner();
     const { authenticateAsync, isPending } = useAuthenticate({
       onMutate: async (params) => {
+        const cfg = await signer?.getConfig();
         if (params.type === "email" && "email" in params) {
-          if (params.emailMode === "magicLink") {
+          const emailMode = legacyEmailMode
+            ? legacyEmailMode === "magicLink"
+              ? "MAGIC_LINK"
+              : "OTP"
+            : cfg?.email.mode;
```

Comment:
nit: I would probably swap the order of priority here to this since if its present on the server and legacyMode doesn't match you're gonna get an error anyways when you call the server

```suggestion
          const emailMode = cfg?.email.mode ? cfg.email.mode : legacyEmailMode
            ? legacyEmailMode === "magicLink"
              ? "MAGIC_LINK"
              : "OTP";
```